pandas/core/common.py

```diff
@@ -508,18 +508,14 @@ def get_rename_function(mapper):
     Returns a function that will map names/labels, dependent if mapper
     is a dict, Series or just a function.
     """
-    if isinstance(mapper, (abc.Mapping, ABCSeries)):
 
-        def f(x):
-            if x in mapper:
-                return mapper[x]
-            else:
-                return x
-
-    else:
-        f = mapper
```

Comment:
pyright doesn't like re-declaring a function in addition to variables. For example, mypy will not allow this variable re-declaration

```py
if flag:
    a: int = 1
else:
    a: int = 1  # error: Name "a" already defined on line 2  [no-redef]
```

pyright also applies this concept to function declarations (mypy does not).